widget_button.go

```diff
@@ -57,6 +66,9 @@ func (w *ButtonWidget) Update(dev *streamdeck.Device) error {
 	height := size - (margin * 2)
 	img := image.NewRGBA(image.Rect(0, 0, size, size))
 
+	if w.icon != nil && w.flatten {
```

Comment:
I think this should be done in `NewButtonWidget`, so it only gets flattened once, not on every call of `Update`.

Reply:
In case we move the call of `flattenImage` into `NewButtonWidget` this wont apply for the weather widget.
Since the weather image does only construct a `ButtonWidget` and not call `NewButtonWidget`